js/client.js

```diff
@@ -843,10 +843,27 @@
 						if (data === 'nonexistent' && Config.server.id && roomid.slice(0, 7) === 'battle-') {
 							var replayid = roomid.slice(7);
 							if (Config.server.id !== 'showdown') replayid = Config.server.id + '-' + replayid;
-							var replayLink = 'http://replay.pokemonshowdown.com/' + replayid;
-							errormessage += '\n\nYou might want to try the replay: ' + replayLink;
+							var replayFound = false;
+							var replayLink = 'https://replay.pokemonshowdown.com/' + replayid;
+							$.ajax(replayLink + '.json', {dataType: 'json'}).done(function (replay) {
+								if (replay) {
+									var title = Tools.escapeHTML(replay.p1) + ' vs. ' + Tools.escapeHTML(replay.p2);
+									app.receive('>battle-' + replayid + '\n|init|battle\n|title|' + title + '\n' + replay.log);
+									app.receive('>battle-' + replayid + '\n|expire|<a href=' + replayLink + ' target="_blank">Open replay in new tab</a>');
+									replayFound = true;
+								} else {
+									errormessage += '\n\nResponse received, but no data.';
+								}
+							}).fail(function () {
+								errormessage += '\n\nYou might want to try the replay: ' + replayLink;
+							}).always(function () {
```

Comment:
I don't think there's a need for an `always` function; just handle the two conditions separately.